### PR TITLE
Implemented Exchange Sort Algorithm

### PR DESCRIPTION
**Pull request by GervinFung**
Good evening sir, I am Gervin, I took some time and implemented a very beginner-friendly and intuitive algorithm
- Exchange Sort

I am looking forward to your answer and maybe the merge of my commits to your original code.
Thank you in advance.
Gervin Fung

### CODE REVIEW

File: test/sorts.jl [hunk 1]
```diff
@@ -14,4 +14,7 @@
     x=[3,5,1,4,2]
     QuickSort!(x)
     @test x == [1,2,3,4,5]
+    x=[3,5,1,4,2]
```
Comment (Azzaare) [inline test/sorts.jl:17]: You probably don't need to redefine `x` here ;)

**Review by Azzaare — DISMISSED** [review]
Seems good to me, as long as the CI passes, it is good to be merged.

**Review by Azzaare — DISMISSED** [review]

**Review by Azzaare — APPROVED** [review]

**Review by Azzaare — APPROVED** [review]

File: src/sorts/exchange_sort.jl [hunk 2]
```diff
@@ -0,0 +1,12 @@
+function ExchangeSort!(arr::Vector{T})where T
```
Comment (Panquesito7) [inline src/sorts/exchange_sort.jl:1]: Missing documentation for this function and algorithm.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: src/sorts/exchange_sort.jl (hunk 2 above)
Comment (GervinFung) [reply src/sorts/exchange_sort.jl:1]: Hi, thanks you giving me feedback, do I create a new PR by adding documentation?

File: src/sorts/exchange_sort.jl [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+4. Returns [1, 2, 3, 4, 5]
+
+Program to sort an array
+
+# Example
+```julia
+ExchangeSort([5, 4, 3, 2, 1]) # returns [1, 2, 3, 4, 5]
+ExchangeSort([1, 2, 3, 4, 5]) # returns [1, 2, 3, 4, 5]
+ExchangeSort([1, 2, 3, 4, 5]) # returns 123.1
+ExchangeSort(['5', '4', '3', '2', '1']) # returns ['1', '2', '3', '4', '5']
```
Comment (Panquesito7) [inline src/sorts/exchange_sort.jl:21]:
```suggestion
ExchangeSort(['5', '4', '3', '2', '1']) # returns ['1', '2', '3', '4', '5']

Contributed By:- [Gervin Fung](https://github.com/GervinFung)
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Great work! Almost there. 😄

**Review by Azzaare — APPROVED** [review]

File: src/sorts/exchange_sort.jl (hunk 3 above)
Comment (GervinFung) [reply src/sorts/exchange_sort.jl:21]: 😄😄😄 didn't notice the return 123.1

**Pull request merged**